assets/js/components/KeyMetrics/ConfirmSitePurposeChangeModal.js

```diff
@@ -115,14 +122,50 @@ function ConfirmSitePurposeChangeModal( {
 		setUIValues,
 	] );
 
-	const { saveUserInputSettings } = useDispatch( CORE_USER );
+	const userInputPurposeConversionEvents = useSelect( ( select ) => {
+		const isGA4Connected =
+			select( CORE_MODULES ).isModuleConnected( 'analytics-4' );
+
+		if ( ! isGA4Connected ) {
```

Comment:
Yes it is anticipated it might be undefined if not resolved yet, it is safe to have empty array in the meantime until detected events are resolved which will also be an array